### ISSUE TITLE
Terminal tab closing during antigen theme invoked by .autoenv.zsh

### ISSUE THREAD

**Issue by shoeffner** [zsh-users/antigen#519]
OS: Mac OS Sierra 10.12.4
zsh --version: zsh 5.3.1 (x86_64-apple-darwin16.3.0)
antigen version: Antigen v2.0.1

I am using [Tarrasch/zsh-autoenv](https://github.com/Tarrasch/zsh-autoenv) mostly to enable python and conda environments. However, in one directory I want to have a different zsh-theme, so my `.autoenv.zsh` looks like this:

```
antigen bundle emoji
antigen theme https://gist.github.com/2a0d83c9e78c51c88fd91524d0bbdd0b.git mini-zoo
source activate zooenv
```

When entering the appropriate directory my Terminal (the default Mac Terminal.app) I can see some programs flushing in the title bar (sed, grep, and a couple others I can't distinguish quick enough) before **the terminal tab eventually closes without further notice**. The log only contains the information about the last updates:

```
master
aae3388491c2312c4efb2e86bcb999927bb2900e
Your branch is up-to-date with 'origin/master'.
Already up-to-date.
aae3388491c2312c4efb2e86bcb999927bb2900e
master
462779629a0c66e652d3bf9f8a1f18330f379ee7
Your branch is up-to-date with 'origin/master'.
From https://github.com/zsh-users/zsh-syntax-highlighting
   4627796..ad522a0  master     -> origin/master
Updating 4627796..ad522a0
Fast-forward
 zsh-syntax-highlighting.zsh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)
ad522a091429ba180c930f84b2a023b40de4dbcc
```

When removing one or another line it turned out that **`antigen theme`** is the culprit. However, when executed manually in the order as written in the .autoenv.zsh file, it works perfectly.
I tried protecting did, but it did not help either (http://stackoverflow.com/a/8811800/3004221):

```
if test "${_ANTIGEN_BUNDLE_RECORD#*mini-zoo.zsh-theme}" = "$_ANTIGEN_BUNDLE_RECORD"
then
    antigen bundle emoji
    antigen theme https://gist.github.com/2a0d83c9e78c51c88fd91524d0bbdd0b.git mini-zoo
fi
source activate zooenv
```

I also tried `antigen reset` and `antigen cleanup` which at least causes `antigen theme` to download the gist again.

This broke with the most recent update I received, though I assume I did not update the last two weeks or so. It's also not really important, I just don't switch to the theme now, but I thought it might be worth mentioning as it might be related to something breaking other things as well.

For completeness:

**`.antigenrc`**
```
antigen use oh-my-zsh

antigen bundles <<EOBUNDLES
    git
    pip
    python
    Tarrasch/zsh-autoenv
    vi-mode
    wd
    zsh-users/zsh-syntax-highlighting
    zsh-users/zsh-history-substring-search
    zsh-users/zsh-autosuggestions
EOBUNDLES

antigen theme shoeffner/dotfiles fae.zsh-theme

antigen apply
```

**`.zshrc` (snippet)**
```
source /usr/local/opt/antigen/share/antigen/antigen.zsh
antigen init ${HOME}/.antigenrc
autoload -U +X compinit && compinit
autoload -U +X bashcompinit && bashcompinit
```

**`$_ANTIGEN_BUNDLE_RECORD`**
```
https://github.com/robbyrussell/oh-my-zsh.git lib plugin true https://github.com/robbyrussell/oh-my-zsh.git plugins/git plugin true https://github.com/robbyrussell/oh-my-zsh.git plugins/pip plugin true https://github.com/robbyrussell/oh-my-zsh.git plugins/python plugin true https://github.com/Tarrasch/zsh-autoenv.git / plugin true https://github.com/robbyrussell/oh-my-zsh.git plugins/vi-mode plugin true https://github.com/robbyrussell/oh-my-zsh.git plugins/wd plugin true https://github.com/zsh-users/zsh-syntax-highlighting.git / plugin true https://github.com/zsh-users/zsh-history-substring-search.git / plugin true https://github.com/zsh-users/zsh-autosuggestions.git / plugin true https://github.com/shoeffner/dotfiles.git fae.zsh-theme theme true
```

**Comment by desyncr** [comment]
Thanks for the detailed report. I never used zsh-autoenv so I'll have to take a look at this.

**Comment by desyncr** [comment]
Issue introduced [here](https://github.com/zsh-users/antigen/issues/494).

**Comment by desyncr** [comment]
@shoeffner You may switch to `2.0.0` as a workaround until we release a fix (hopefully for `v2.0.2`)

**Comment by desyncr** [comment]
Should be fixed with #520 if you can take a look!

**Comment by shoeffner** [comment]
Thanks for the very quick reply. I checked your changes out and the enter works fine again.

However, when leaving the directory with the following `.autoenv_leave.zsh` file:

```
if test "${_ANTIGEN_BUNDLE_RECORD#*fae.zsh-theme}" = "$_ANTIGEN_BUNDLE_RECORD"
then
    antigen theme shoeffner/dotfiles fae.zsh-theme
fi
source deactivate zooenv
```

It keeps the theme.
If I add `echo "Changing theme..."` before the `antigen theme` call, I am running into an endless loop of:

```
(eval):1: command not found: Changing
```

I additionally can observe that conda deactivates the environment again and again, so there seems to be some looping going on.
It is a bit surprising that it works into one way but not the other now, even though I am using basically the same commands.

If I remove all commands but the following:
```
# .autoenv.zsh
antigen bundle emoji
antigen theme https://gist.github.com/2a0d83c9e78c51c88fd91524d0bbdd0b.git mini-zoo
```
```
# .autoenv_leave.zsh
antigen theme shoeffner/dotfiles fae.zsh-theme
```
It again works into one direction (enter) but when leaving the directory I end up with this (just once this time):
`(eval):1: command not found: Installing`

```
cd zooenv
🐠  % cd ..   
(eval):1: command not found: Installing
cd zooenv
🐗  % 
```

Leaving me without any style (as shown above) at all.

**Comment by desyncr** [comment]
Thanks again for the detailed report. Can confirm there are still issues with autoenv and theme switching.

A workaround I believe is using `antigen bundle` rather than `antigen theme`.

You may have to call `-antigen-theme-reset-hooks` (https://github.com/zsh-users/antigen/blob/develop/src/commands/theme.zsh#L46) in order to clear hooks set up by themes in order to avoid them conflicting with each other.

**Comment by desyncr** [comment]
Should be fixed with latest changes in #520. Basically autoenv chpwd hook messed up with theme loading.
